### PR TITLE
proxy config: clarify concurrency docs

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/api/issues/1783

### CODE REVIEW

File: mesh/v1alpha1/proxy.proto [hunk 1]
```diff
@@ -456,9 +456,9 @@ message ProxyConfig {
   int32 stat_name_length = 15;
 
   // The number of worker threads to run.
-  // If unset, this will be automatically determined based on CPU requests/limits.
-  // If set to 0, all cores on the machine will be used.
-  // Default is 2 worker threads.
```
Comment (craigbox) [inline mesh/v1alpha1/proxy.proto:461]: Do we want to lose this fact?
#1783 suggests it's still true

Comment (howardjohn) [reply mesh/v1alpha1/proxy.proto:461]: It is sorta true sort of not... https://github.com/istio/api/issues/1783#issuecomment-2111262691.

File: mesh/v1alpha1/proxy.proto [hunk 2]
```diff
@@ -456,9 +456,9 @@ message ProxyConfig {
...
+  // If unset, which is recommended, this will be automatically determined based on CPU requests/limits.
```
Comment (hzxuzhonghu) [inline mesh/v1alpha1/proxy.proto:459]: Based on pod cpu

**Review by zirain — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

File: mesh/v1alpha1/proxy.proto (hunk 2 above)
Comment (rehevkor5) [reply mesh/v1alpha1/proxy.proto:459]: > based on CPU requests/limits

How, though?

Comment (rehevkor5) [reply mesh/v1alpha1/proxy.proto:459]: Based on https://github.com/istio/istio/blob/af58e9e09598f8181ae217ea886c747a288f0a2c/pilot/cmd/pilot-agent/config/config.go#L64 it looks like CPU Request doesn't have anything to do with it. If CPU Limit is set, it looks like they take the ceiling (round up) of the CPU Limit and use that as the concurrency value. It's unclear what happens if CPU Limit is not set.

Comment (howardjohn) [reply mesh/v1alpha1/proxy.proto:459]: If its not set it uses 1 thread per core on the machine

**Pull request merged**